crates/eth_compatibility/src/lib.rs

```diff
@@ -1,150 +0,0 @@
-    fn from(bytes: [u8; 20]) -> Self {
-        Self(bytes)
-    }
-}
-
-impl ECDSAPublicKey {
-    /// Returns Ethereum address from the ECDSA compressed public key.
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// use ink_eth_compatibility::{ECDSAPublicKey, EthereumAddress};
-    /// let pub_key: ECDSAPublicKey = [
-    ///     2, 121, 190, 102, 126, 249, 220, 187, 172, 85, 160,  98, 149, 206, 135, 11,
-    ///     7,   2, 155, 252, 219,  45, 206,  40, 217, 89, 242, 129,  91,  22, 248, 23,
-    ///     152,
-    /// ].into();
-    ///
-    /// let EXPECTED_ETH_ADDRESS: EthereumAddress = [
-    ///     126, 95, 69, 82, 9, 26, 105, 18, 93, 93, 252, 183, 184, 194, 101, 144, 41, 57, 91, 223
-    /// ].into();
-    ///
-    /// assert_eq!(pub_key.to_eth_address().as_ref(), EXPECTED_ETH_ADDRESS.as_ref());
-    /// ```
-    // We do not include this function on Windows, since it depends on `libsecp256k1`,
-    // which is incompatible with Windows.
-    // We have https://github.com/paritytech/ink/issues/1068 for removing this
-    // dependency altogether.
-    #[cfg(not(target_os = "windows"))]
-    pub fn to_eth_address(&self) -> EthereumAddress {
-        use ink_env::hash;
-        use libsecp256k1::PublicKey;
-
-        // Transform compressed public key into uncompressed.
-        let pub_key = PublicKey::parse_compressed(&self.0)
-            .expect("Unable to parse the compressed ECDSA public key");
-        let uncompressed = pub_key.serialize();
-
-        // Hash the uncompressed public key by Keccak256 algorithm.
-        let mut hash = <hash::Keccak256 as hash::HashOutput>::Type::default();
-        // The first byte indicates that the public key is uncompressed.
-        // Let's skip it for hashing the public key directly.
-        ink_env::hash_bytes::<hash::Keccak256>(&uncompressed[1..], &mut hash);
-
-        // Take the last 20 bytes as an Address
-        let mut result = EthereumAddress::default();
-        result.as_mut().copy_from_slice(&hash[12..]);
-
-        result
-    }
-
-    /// Returns the default Substrate's `AccountId` from the ECDSA compressed public key.
-    /// It hashes the compressed public key with the `blake2b_256` algorithm like in substrate.
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// use ink_eth_compatibility::ECDSAPublicKey;
-    /// let pub_key: ECDSAPublicKey = [
-    ///     2, 121, 190, 102, 126, 249, 220, 187, 172, 85, 160,  98, 149, 206, 135, 11,
-    ///     7,   2, 155, 252, 219,  45, 206,  40, 217, 89, 242, 129,  91,  22, 248, 23,
-    ///     152,
-    /// ].into();
-    ///
-    /// const EXPECTED_ACCOUNT_ID: [u8; 32] = [
-    ///     41, 117, 241, 210, 139, 146, 182, 232,  68, 153, 184, 59,   7, 151, 239, 82,
-    ///     53,  85,  62, 235, 126, 218, 160, 206, 162,  67, 193, 18, 140,  47, 231, 55,
-    /// ];
-    ///
-    /// assert_eq!(pub_key.to_default_account_id(), EXPECTED_ACCOUNT_ID.into());
-    pub fn to_default_account_id(
```

Comment:
@athei Hmm, do you see any way how we could keep this function?

Reply:
I think we could make this as a [trait extension](https://github.com/paritytech/substrate/pull/11087/files#diff-9237527ee4beab4bda27f69f53a62a4dc122bb00d92ab22d21701e6d21813b0dR35) for `Public` in `frame_support`, and then add a `seal_` function to `pallet_contracts` API, just like it's done for `to_eth_address()`. Question is: is this function really needed though, what are use cases for it?

Reply:
>Question is: is this function really needed though, what are use cases for it?

@xgreenx Do you have something to add here? Is this function maybe used by OpenBrush?

Reply:
This function doesn't depend on any async cryptography. It just hashes some bytes which relies on pallet contracts. We could just copy it to `ink_env`. However, I would just remove it as hash functions are already exposed.